test/CodeGen/AVR/umul-with-overflow.ll

```diff
@@ -0,0 +1,11 @@
+; RUN: llc < %s -march=avr | FileCheck %s
+
+define i1 @unsigned_multiplication_did_overflow(i8, i8) unnamed_addr {
+; CHECK-LABEL: unsigned_multiplication_did_overflow:
+entry-block:
```

Comment:
The assembly:

```asm
unsigned_multiplication_did_overflow:   ; @unsigned_multiplication_did_overflow
; BB#0:                                 ; %entry-block
	mul	r24, r22
	mov	r25, r1
	eor	r1, r1
	ldi	r24, 1
	cpi	r25, 0
	brne	LBB0_2
; BB#1:                                 ; %entry-block
	ldi	r24, 0
LBB0_2:                                 ; %entry-block
	ret
```

This one is straight-forward - do the multiply and then check to see if the high byte is zero. There's some small inefficiencies. What general shape of tests would you like to see?